app/src/parser_impl.c

```diff
@@ -251,147 +251,84 @@ parser_error_t _matchScriptType(uint8_t scriptHash[32], script_type_e *scriptTyp
 
-    if (
-        (MEMCMP(TEMPLATE_HASH_TH19_DELEGATE_NEW_TOKENS_TESTNET, buffer, 64) == 0) ||
-        (MEMCMP(TEMPLATE_HASH_TH19_DELEGATE_NEW_TOKENS_MAINNET, buffer, 64) == 0)) {
-        *scriptType = SCRIPT_TH19_DELEGATE_NEW_TOKENS;
-        return PARSER_OK;
-    }
+        {SCRIPT_TH19_DELEGATE_NEW_TOKENS, TEMPLATE_HASH_TH19_DELEGATE_NEW_TOKENS_TESTNET},
+        {SCRIPT_TH19_DELEGATE_NEW_TOKENS, TEMPLATE_HASH_TH19_DELEGATE_NEW_TOKENS_MAINNET},
 
-    if (
-        (MEMCMP(TEMPLATE_HASH_TH20_RESTAKE_UNSTAKED_DELEGATED_TOKENS_TESTNET, buffer, 64) == 0) ||
-        (MEMCMP(TEMPLATE_HASH_TH20_RESTAKE_UNSTAKED_DELEGATED_TOKENS_MAINNET, buffer, 64) == 0)) {
-        *scriptType = SCRIPT_TH20_RESTAKE_UNSTAKED_DELEGATED_TOKENS;
-        return PARSER_OK;
-    }
+        {SCRIPT_TH20_RESTAKE_UNSTAKED_DELEGATED_TOKENS, TEMPLATE_HASH_TH20_RESTAKE_UNSTAKED_DELEGATED_TOKENS_TESTNET},
+        {SCRIPT_TH20_RESTAKE_UNSTAKED_DELEGATED_TOKENS, TEMPLATE_HASH_TH20_RESTAKE_UNSTAKED_DELEGATED_TOKENS_MAINNET},
 
-    if (
-        (MEMCMP(TEMPLATE_HASH_TH21_RESTAKE_REWARDED_DELEGATED_TOKENS_TESTNET, buffer, 64) == 0) ||
-        (MEMCMP(TEMPLATE_HASH_TH21_RESTAKE_REWARDED_DELEGATED_TOKENS_MAINNET, buffer, 64) == 0)) {
-        *scriptType = SCRIPT_TH21_RESTAKE_REWARDED_DELEGATED_TOKENS;
-        return PARSER_OK;
-    }
+        {SCRIPT_TH21_RESTAKE_REWARDED_DELEGATED_TOKENS, TEMPLATE_HASH_TH21_RESTAKE_REWARDED_DELEGATED_TOKENS_TESTNET},
+        {SCRIPT_TH21_RESTAKE_REWARDED_DELEGATED_TOKENS, TEMPLATE_HASH_TH21_RESTAKE_REWARDED_DELEGATED_TOKENS_MAINNET},
 
-    if (
-        (MEMCMP(TEMPLATE_HASH_TH22_UNSTAKE_DELEGATED_TOKENS_TESTNET, buffer, 64) == 0) ||
-        (MEMCMP(TEMPLATE_HASH_TH22_UNSTAKE_DELEGATED_TOKENS_MAINNET, buffer, 64) == 0)) {
-        *scriptType = SCRIPT_TH22_UNSTAKE_DELEGATED_TOKENS;
-        return PARSER_OK;
-    }
+        {SCRIPT_TH22_UNSTAKE_DELEGATED_TOKENS, TEMPLATE_HASH_TH22_UNSTAKE_DELEGATED_TOKENS_TESTNET},
+        {SCRIPT_TH22_UNSTAKE_DELEGATED_TOKENS, TEMPLATE_HASH_TH22_UNSTAKE_DELEGATED_TOKENS_MAINNET},
 
-    if (
-        (MEMCMP(TEMPLATE_HASH_TH23_WITHDRAW_UNSTAKED_DELEGATED_TOKENS_TESTNET, buffer, 64) == 0) ||
-        (MEMCMP(TEMPLATE_HASH_TH23_WITHDRAW_UNSTAKED_DELEGATED_TOKENS_MAINNET, buffer, 64) == 0)) {
-        *scriptType = SCRIPT_TH23_WITHDRAW_UNSTAKED_DELEGATED_TOKENS;
-        return PARSER_OK;
-    }
+        {SCRIPT_TH23_WITHDRAW_UNSTAKED_DELEGATED_TOKENS, TEMPLATE_HASH_TH23_WITHDRAW_UNSTAKED_DELEGATED_TOKENS_TESTNET},
+        {SCRIPT_TH23_WITHDRAW_UNSTAKED_DELEGATED_TOKENS, TEMPLATE_HASH_TH23_WITHDRAW_UNSTAKED_DELEGATED_TOKENS_MAINNET},
+
+        {SCRIPT_TH24_WITHDRAW_REWARDED_DELEGATED_TOKENS, TEMPLATE_HASH_TH24_WITHDRAW_REWARDED_DELEGATED_TOKENS_TESTNET},
+        {SCRIPT_TH24_WITHDRAW_REWARDED_DELEGATED_TOKENS, TEMPLATE_HASH_TH24_WITHDRAW_REWARDED_DELEGATED_TOKENS_MAINNET},
+        // sentinel, do not remove
+        {0, NULL}
```

Comment:
👌 
Nice, as long as we make sure no other template is `NULL`.

Reply:
The worst case that would happen would be premature cutoff of entries, i.e. app would not recognize some templates. This would be safe from the security perspective